ostd/src/arch/x86/iommu/registers/mod.rs

```diff
@@ -103,20 +103,41 @@ impl IommuRegisters {
         GlobalStatus::from_bits_truncate(self.global_status.read())
     }
 
-    /// Enables DMA remapping with static RootTable
-    pub(super) fn enable_dma_remapping(
+    /// Enables abort-dma mode
+    pub(super) fn enable_abort_dma_mode(&mut self) {
+        // Check if hardware supports abort-dma mode.
+        if !self
+            .read_extended_capability()
+            .flags()
+            .contains(ExtendedCapabilityFlags::ADMS)
+        {
+            return;
+        }
+
+        const ABORT_DMA_MODE: u64 = 0xC00;
+        // SAFETY:
+        // 1. We have checked that the hardware supports abort-dma mode, the value is valid.
+        // 2. According to the manual: "In abort-dma mode, hardware will abort all DMA operations
+        // without the need to set up a root-table with each entry marked as not-present." In this case,
+        // there will be no soundness problems related to DMA (All access is aborted).
+        unsafe {
+            self.enable_dma_with_address_register(ABORT_DMA_MODE);
+        }
+    }
+
+    /// Enables DMA remapping with static RootTable.
+    ///
+    /// # Safety
+    ///
+    /// The user must ensure that all the entry inside the root table is set to not-present.
+    pub(super) unsafe fn enable_dma_remapping(
```

Comment:
In practice, the user will always pass an empty root table, right? And I don't think it's reasonable for the user to pass a non-empty root table (with all entries not-present).

I think we can do some check in this function to ensure the root table is empty. Then, this function can be marked as safe function.